openwisp_users/tests/test_admin.py

```diff
@@ -511,6 +511,119 @@ def test_change_staff_with_group(self):
+        self.assertEqual(r.status_code, 200)
+        self.assertContains(r, f'class="readonly">{user2.username}')
+
+        org_owner.delete()
+        path = reverse(f'admin:{self.app_label}_user_change', args=[user2.pk])
+        r = self.client.get(path)
+        self.assertEqual(r.status_code, 200)
+        self.assertNotContains(r, f'class="readonly">{user2.username}')
+
+    def test_org_owner_can_edit_staff(self):
+        user1 = self._create_user(
+            username="user1", email="email1@mail.com", is_staff=True
+        )
+        user2 = self._create_user(
+            username="user2", email="email2@mail.com", is_staff=True
+        )
+        org = self._get_org()
+        org_user2 = self._create_org_user(user=user2, organization=org, is_admin=True)
+        self._create_org_owner(organization=org, organization_user=org_user2)
+        self._create_org_user(user=user1, organization=org, is_admin=True)
+        group = Group.objects.get(name='Administrator')
+        user1.groups.add(group)
+        user2.groups.add(group)
+        self.client.force_login(user2)
+        params = user1.__dict__
+        params['username'] = 'newuser1'
+        params['groups'] = str(group.pk)
+        params.pop('phone_number')
+        params.pop('_password')
+        params.pop('last_login')
+        params.update(self.add_user_inline_params)
+        params.update(self._additional_params_add())
+        params.update(self._get_edit_form_inline_params(user1, org))
+        path = reverse(f'admin:{self.app_label}_user_change', args=[user1.pk])
+        r = self.client.post(path, params, follow=True)
+        self.assertEqual(r.status_code, 200)
+        self.assertNotContains(r, 'error')
+        user1.refresh_from_db()
+        self.assertEqual(user1.username, 'newuser1')
+
+    def test_org_owner_can_edit_org_owner(self):
+        user = self._create_user(username="user", email="email@mail.com", is_staff=True)
+        org = self._get_org()
+        org_user = self._create_org_user(user=user, organization=org, is_admin=True)
+        group = Group.objects.get(name='Administrator')
+        user.groups.add(group)
+        self._create_org_owner(organization=org, organization_user=org_user)
+        self.client.force_login(user)
+        params = user.__dict__
+        params['username'] = 'newusername'
+        params['groups'] = str(group.pk)
+        params.pop('phone_number')
+        params.pop('_password')
+        params.pop('last_login')
+        params.update(self.add_user_inline_params)
+        params.update(self._additional_params_add())
+        params.update(self._get_edit_form_inline_params(user, org))
+        path = reverse(f'admin:{self.app_label}_user_change', args=[user.pk])
+        r = self.client.post(path, params, follow=True)
+        self.assertEqual(r.status_code, 200)
+        self.assertNotContains(r, 'error')
+        user.refresh_from_db()
+        self.assertEqual(user.username, 'newusername')
+
```

Comment:
I think one important case is missing from the tests: a staff user should be able to edit their own account.
This works now, but someone may inadvertently break this in the future, so if we add a test to ensure this works we prevent this future bug from happening.